src/index.ts

```diff
@@ -0,0 +1,107 @@
+export enum NodeTypes {
+  ELEMENT_NODE = 1,
+  ATTRIBUTE_NODE = 2,
+  TEXT_NODE = 3,
+  CDATA_SECTION_NODE = 4,
+  PROCESSING_INSTRUCTION_NODE = 7,
+  COMMENT_NODE = 8,
+  DOCUMENT_NODE = 9,
+  DOCUMENT_TYPE_NODE = 10,
+  DOCUMENT_FRAGMENT_NODE = 11,
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function isNodeLike(value: any): value is Node {
```

Comment:
I have. In this case you have to add a ton of code to get `unknown` to work correctly. All the other options are `unknown`, but here it makes more sense to use `any`. The problem with `any` is typically that it could be anything and you have to type-narrow it down to make sure you don't do anything wrong. The purpose of this function is to type narrow, so that fits.

Conversely, the purpose of `unknown` is to say that this is basically nothing, so you have to do a bunch of checks to build the object up, and then you have to run your code, which, in our case, does a bunch of checks; it is redundant.

For example, I can't do `if (varOfTypeUnknown) ...` because `varOfTypeUnknown` isn't a boolean, so my first line already requires lots of scaffolding to even run. If I leave it as `any`, then normal JS works. I'm not a big fan of adding a lot of code just to please the type checker. If that is required, then the type checker is getting in the way. If you feel strongly that I should add all the extra code, I will.

Reply:
No, makes sense, but knowing that, we can just disable the eslint rule that complains from the config, instead of in the file.

Reply:
Ideally we'd still have the rule so that we don't use `any` anywhere else. If you'd prefer the rule disabled, I don't mind that either. I don't expect this repo will get a lot of churn in anything the annual dependency updating.